GameProject/Assets/TurnBasedAssets/Scripts/PathFinding/PathFinder.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using TurnBasedAssets.Scripts.Interface;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace TurnBasedAssets.Scripts.PathFinding
+{
+    public class PathFinder : IPathFinder
+    {
+        public List<Vector3> pathToFollow = new List<Vector3>();
+
+        private float _2dMaxDistance = 1;
+        public class Location
+        {
+            public Location Parent { get; set; }
+            public Vector3 PositionInWorld { get; }
+
+            private float G => Parent?.G + 1 ?? 0;
+            private float H { get; }
+            public float F => Mathf.Round(G + H);
+
+
+            public Location(Vector3 position, Vector3 destination, Location parent)
+            {
+                Parent = parent;
+                PositionInWorld = position;
+                H = Vector3.Distance(position, destination);
+            }
+        }
+
+        public IEnumerator FindPath(Vector3 startPosition, Vector3 targetPosition, bool is3D, float movementRadius, Action<IEnumerable<Vector3>> onCompletion)
```

Comment:
Consider refactoring this into multiple methods. While the code fits conventions its still a little hard to read with all the functionality this one method has.